Apps Overview (#242)
farm-ng/amiga-dev-kit: website/docs/apps/controller_app/README.md

---

> id: controller-app

Reviewer: ```suggestion
id: autoplot-app
```
+ where applicable & rename file `autoplot_app/README.md`

Reviewer 2: Let's do a follow-up PR to scrub controller references.

Autonomy will not engage if the filter has not converged.
- **GPS**: A visual indicator showing if the GPS service is active and functional.
- **WiFi**: This shows if the robot is connected to Wifi, which is necessary for autonomous operations.
- **Battery**: Displays the current battery level.
Always monitor this, especially before starting long tasks.

<div align="center">
  <img src="https://github.com/farm-ng/amiga-dev-kit/assets/39603677/044304e6-1ae5-4724-845f-da4a9ebe4def"
  alt="image" />
</div>

## Tips for Optimal Autonomy

- Before engaging in autonomous operations, always perform a safety check of the environment.
- When operating the robot via remote keyboard control in the web app, ensure a stable and
responsive network connection.
- Monitor the robot's battery status via the Autoplot app to avoid interruptions in longer operations.

## Troubleshooting

If you encounter issues with autonomy or control:

- Verify all status indicators in the Autoplot app's top bar are active and showing the expected states.
- Check the [**Filter App**](/docs/apps/filter_app)
to confirm that calibration and GPS settings are up-to-date.

> - If the robot does not respond to WASD commands, ensure that no other tasks are running and that

Reviewer 2: ```suggestion
- If the robot does not respond to WAD keyboard commands, ensure that no other tasks are running and that
```

Reviewer 2: I'll go ahead and commit this, and merge.

---
id: controller-app
title: Autoplot App
---

# Autoplot App Guide

The Autoplot app is an advanced control interface for your Amiga, offering multiple modes of
operation such as teach-and-repeat, WASD control, and remote access through a web browser.

## How to Use the Autoplot App

> ### Teach-and-Repeat

Author: Done here: 7bf35646

The teach-and-repeat functionality allows you to manually drive the robot along a desired path
and then have the robot repeat the path autonomously.

1. Click "Record" to start mapping your path.
2. Drive the robot as desired.
3. Click "Save" and name your track to use it later.
4. Load the track when needed and click "Start Track" to initiate autonomous follow.

### Remote Access via Web Browser

The Autoplot is a web-app, and can be access through any web browser, such as Google Chrome.

> To view it in your browser, navigate to: `<element-vegetable>:8008`

Reviewer 2: We should note the prereq that they've [configured cross-network access](https://amiga.farm-ng.com/docs/ssh/#recommended-configure-cross-network-access) and added their web browser device to their Tailscale network.

Customers will need to use the FQDN form `[element-fruit].[tailnet-name].ts.net:8008`.

Author: Note added here: 7bf35646

:::note NOTE
Make sure to rename `<element-vegetable>` with your actual robot name.
:::

> This app runs persistently in the background, even when not active on the robot's display,
> ensuring that you can control your Amiga from any web browser at any time.

Reviewer 2: ```suggestion
You can control your Amiga from a web browser at any time, even when the Autoplot app is not active on the Brain display.
```

"app runs persistently in the background" is a bit of implementation detail, and might not be the best way to describe it in the future.

### WASD Control

WASD is possible on web browsers.
When your robot is in `automode`, you can take direct control with simple keyboard commands.

- **W**: Move forward
- **A**: Turn counterclockwise in place
- **D**: Turn clockwise in place
- **S**: Disabled for safety

Pressing these keys in your web browser will send commands directly to the robot, allowing for
immediate control.

:::info
Using WASD control will override any ongoing autonomous tasks.
:::

> ## Autonomy Requirements

Author: Good point. I provided more information here: 7bf35646

Before starting any operation, make sure to check the following:

- [ ] The robot is calibrated through the [**Filter App**](/docs/apps/filter_app).
- [ ] The GPS antenna offsets are correctly set in the [**Filter App**](/docs/apps/filter_app).

> - [ ] Ensure the robot is connected to an RTK base station for enhanced GPS accuracy.

Author: Done here: 7bf35646
